# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContextExtensions.cpp
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLContextExtensions.cpp
--- a/dom/canvas/WebGLContextExtensions.cpp	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLContextExtensions.cpp	Wed Jan 08 22:19:23 2020 +0000
@@ -9,8 +9,6 @@
 #include "mozilla/dom/BindingDeclarations.h"
 #include "mozilla/dom/ToJSValue.h"
 #include "mozilla/EnumeratedRange.h"
-#include "mozilla/Preferences.h"
-#include "nsContentUtils.h"
 #include "nsString.h"
 #include "WebGLContextUtils.h"
 #include "WebGLExtensions.h"
@@ -211,7 +209,7 @@
         case WebGLExtensionID::Max:
           break;
       }
-      MOZ_ASSERT_UNREACHABLE("illegal extension enum");
+      MOZ_CRASH("illegal extension enum");
     }();
     MOZ_ASSERT(extSlot);
     RequestExtension(ext);
@@ -225,8 +223,10 @@
 
 bool WebGLContext::IsExtensionSupported(WebGLExtensionID ext) const {
   switch (ext) {
+    case WebGLExtensionID::MOZ_debug:
+    case WebGLExtensionID::WEBGL_debug_renderer_info:
+    case WebGLExtensionID::WEBGL_debug_shaders:
     case WebGLExtensionID::WEBGL_lose_context:
-    case WebGLExtensionID::MOZ_debug:
       // Always supported.
       return true;
 
@@ -328,13 +328,6 @@
     case WebGLExtensionID::WEBGL_compressed_texture_s3tc_srgb:
       return WebGLExtensionCompressedTextureS3TC_SRGB::IsSupported(this);
 
-    case WebGLExtensionID::WEBGL_debug_renderer_info:
-      return Preferences::GetBool("webgl.enable-debug-renderer-info", false) &&
-             !mResistFingerprinting;
-
-    case WebGLExtensionID::WEBGL_debug_shaders:
-      return !mResistFingerprinting;
-
     case WebGLExtensionID::WEBGL_depth_texture:
       return WebGLExtensionDepthTexture::IsSupported(this);
 